src/routes.tsx

```diff
@@ -9,13 +9,16 @@
 export const privateRoutes = [
   'minndla',
   'minndla/folders',
+  'minndla/folders/preview/:folderId',
+  'minndla/folders/preview/:folderId/:subFolderId',
+  'minndla/folders/preview/:folderId/:subFolderId/:resourceId',
```

Comment:
Går det ikke an å ha `/:folderId/:resourceId`?

Reply:
Tja, det går kanskje an. Det er hovedsakelig 2 grunner til at jeg ikke gjorde det:
- Disse stiene er strukturert likt som under /folder, for ikke å komplisere vedlikehold
- Jeg er prinsipielt skeptisk til å ha endepunkt med 2 forskjellige dynamiske variabler (/preview/:folderId/:subFolderId og /preview/:folderId/:resourceId). URL blir kortere, men jeg synes ikke det veier opp for den ekstra risikoen for bugs.

Reply:
Tror ikke helt du skjønner hva jeg mener. Ta for eksempel denne:
http://localhost:3000/minndla/folders/preview/b8d194ca-78c2-4b50-9d2b-497774e8608e/b8d194ca-78c2-4b50-9d2b-497774e8608e/2f0416f1-cf48-4cbe-9a64-97276ad1bea0

Dette er en forhåndsvisning av en mappe som er lagret på toppnivå hos Frank foreleser. Mappen har igjen ressurser på sitt rotnivå. URL'en jeg lenket til lenger opp er til en av disse ressursene. Merk at folder-id'en gjentar seg selv to ganger før ressurs-ID'en kommer. Hvis vi må duplisere mappe-id'en for å komme i mål mener jeg at vi bør endre noe.

Ser nå at det er sånn i test også, men det burde det da vel ikke være? Halvparten av meningen med URL'ene i Min NDLA var at de skulle være kortere med færre ledd. Hvis vi dupliserer leddene gjør vi jo akkurat samme feilene som vi har prøvd å lære av fra fagsidene våre.

Reply:
Har antatt at det var med overlegg, nettopp fordi react router ikke har noen måte å bestemme hvilken path som er riktig hvis det er mer enn 1 tilgjenglig valg med dynamisk innhold. Jeg kan se 2 mulige måter å gjøre det på:
- slå sammen folders og preview i lenka (de MÅ ikke være under hverandre), slik at det f.eks. istedet er /minndla/pre-folder/... <- Men det sparer kun 1 ledd, og ikke mange tallene, og løser vel egentlig ikke problemed med duplikat ID
- Legge til et ledd mellom ID 1 eller 2 som sier om det er resource eller folder. Lenka ville da f.eks. bli /minndla/folders/preview/......./f/........./........ eller /minndla/folders/preview/....../r/......

Alternativ 2 ville spart det duplikate leddet, men introdusert et ekstra ledd. Kombinasjonen av alternativ 1 og 2 ville gjort at antall ledd ville blitt det samme, men duplikatet ville vært borte. Såvidt jeg fant ut, 1 ledd ikke være slått sammen av både en dynamisk og en statisk del.

Vi kan naturligvis ellers lage vår egen parsing på et ledd, og la det leddet rendere forskjellig avhengig av hva URL inneholder. F.eks. hvis URL er /minndla/folders/preview/:folderId/:subId/ så kunne vi latt :subId være på format f:id eller r:id, sjekke om første tegn er f eller r (ev sjekke om det foran : er resource eller folder, om vi vil være eksplisitte).

Uansett, er det mulig at en slik addressering av lenkeoppbygging bør være en egen PR?

Reply:
Det kan sjå ut som om dobbel rotfolder-id har vore slik ganske lenge ja. Men kan det henge sammen med at vi har støtte for x antall nivå på undermapper? At vi dermed må ha både rot og parent, sjølv om disse er like i dei aller fleste tilfeller.